msbuild/Xamarin.iOS.Tasks.Core/Tasks/MTouchTaskBase.cs

```diff
@@ -527,16 +565,36 @@ protected override string GenerateCommandLineCommands ()
 				};
 
 				for (int i = 0; i < extraArgs.Length; i++) {
-					if (extraArgs[i] == "-gcc_flags" || extraArgs[i] == "--gcc_flags") {
+					var argument = extraArgs[i];
+					int startIndex = 0;
+					int endIndex;
+
+					while (argument.Length > startIndex && argument[startIndex] == '-')
+						startIndex++;
+
+					endIndex = startIndex;
+					while (endIndex < argument.Length && argument[endIndex] != '=')
+						endIndex++;
+
+					if (startIndex > 0)
+						argument = argument.Substring (startIndex, endIndex - startIndex);
```

Comment:
instead of `Substring`, which will allocate memory, you have everything needed to call `String.CompareOrginal`